subprojects/kotlin-dsl-tooling-builders/src/crossVersionTest/groovy/org/gradle/kotlin/dsl/tooling/builders/r54/KotlinBuildScriptModelCrossVersionSpec.groovy

```diff
@@ -0,0 +1,574 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.kotlin.dsl.tooling.builders.r54
+
+import groovy.transform.CompileStatic
+
+import org.gradle.integtests.tooling.fixture.TargetGradleVersion
+import org.gradle.test.fixtures.file.LeaksFileHandles
+
+import org.gradle.kotlin.dsl.tooling.builders.AbstractKotlinScriptModelCrossVersionTest
+
+import org.hamcrest.Matcher
+
+import static org.hamcrest.CoreMatchers.allOf
+import static org.hamcrest.CoreMatchers.equalTo
+import static org.hamcrest.CoreMatchers.not
+import static org.hamcrest.CoreMatchers.hasItem
+import static org.hamcrest.CoreMatchers.hasItems
+import static org.junit.Assert.assertThat
+import static org.junit.Assert.assertTrue
+
+
+@TargetGradleVersion(">=5.4")
+class KotlinBuildScriptModelCrossVersionSpec extends AbstractKotlinScriptModelCrossVersionTest {
+
+    def "can fetch buildSrc classpath in face of compilation errors"() {
+
+        given:
+        withBuildSrc()
+
+        and:
+        withDefaultSettings()
+        withBuildScript("""
+            val p =
+        """)
+
+        expect:
+        assertContainsBuildSrc(canonicalClassPath())
+    }
+
+    def "can fetch buildSrc classpath in face of buildscript exceptions"() {
+
+        given:
+        withBuildSrc()
+
+        and:
+        withDefaultSettings()
+        withBuildScript("""
+            buildscript { TODO() }
+        """)
+
+        expect:
+        assertContainsBuildSrc(canonicalClassPath())
+    }
+
+    def "can fetch buildscript classpath in face of compilation errors"() {
+
+        given:
+        projectDir.createFile("classes.jar")
+        settingsFile << ""
+        buildFileKts << """
+            buildscript {
+                dependencies {
+                    classpath(files("classes.jar"))
+                }
+            }
+
+            val p =
+        """.stripIndent()
+
+        expect:
+        assertClassPathContains(
+            file("classes.jar"))
```

Comment:
⭕️ Let's take the opportunity to close all multiline closing brackets on a separate line, i.e.

```
assertClassPathContains(
    file("classes.jar")
)
```